sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -159,6 +161,23 @@ object RapidsPluginUtils extends Logging {
         logWarning(s"$numThreadsKey is set to $numThreads.")
       }
     }
+
+    // If spark.task.resource.gpu.amount is larger than 
+    // (spark.executor.resource.gpu.amount / spark.executor.cores) then GPUs will be the limiting
+    // resource for task scheduling.
+    if (conf.contains(TASK_GPU_AMOUNT_KEY) && conf.contains(EXECUTOR_GPU_AMOUNT_KEY)) {
+      val taskGpuAmountSetByUser = conf.get(TASK_GPU_AMOUNT_KEY).toDouble
+      // get worker's all cores num if spark.executor.cores is not set explicitly
+      lazy val workerAllCores = Runtime.getRuntime.availableProcessors.toString()
```

Comment:
Why is this marked lazy?  It will be immediately evaluated on the next line to pass its value to `conf.get`.

Reply:
Yes the lazy won't work, updated.